deepspeed/runtime/pipe/module.py

```diff
@@ -562,13 +562,28 @@ def ckpt_layer_path_list(self, ckpt_dir, local_layer_idx):
         return ckpt_files
 
     def save_state_dict(self, save_dir, checkpoint_engine):
-        if self._grid.data_parallel_id != 0:
-            return
+        # temporary, guessing there is already a function like this that can be used instead
+        def get_start_end(count, rank, numranks):
+            num, remainder = divmod(count, numranks)
+            if rank < remainder:
+                start = (num + 1) * rank
+                end = start + num + 1
+            else:
+                start = (num + 1) * remainder + num * (rank - remainder)
+                end = start + num
+            return start, end
```

Comment:
@tjruwase , do you already have a function like this one that I could call instead?

Reply:
Would one of these two work?

1. https://github.com/microsoft/DeepSpeed/blob/80b10d0c6985fb1792a044678c3c5c306cc0922e/deepspeed/runtime/utils.py#L575
2. https://github.com/microsoft/DeepSpeed/blob/80b10d0c6985fb1792a044678c3c5c306cc0922e/deepspeed/runtime/utils.py#L641

Reply:
Yes, thanks.  I've updated this to use ``partition_uniform`` and dropped my ``get_start_end`` function.

Reply:
> > @adammoody, it will be great to make this a configuration option through ds_config. I am thinking of the right way, but please let me know if you have some thoughts on this.
> 
> Sure. Some of the lines that I have dropped could instead be conditioned on a new ds_config option.

Tto disable this feature, I think we could just set ``start,end`` appropriately.
```
num_layers = len(self.forward_funcs)
if args.dp_parallel_write:
  # divide layers evenly among data parallel ranks
  offsets = ds_utils.partition_uniform(num_layers, dp_size)
  start, end = offsets[dp_rank], offsets[dp_rank+1]
else:
  # assign all layers to data parallel rank 0
  if dp_rank == 0:
    start, end = 0, num_layers
  else:
    start, end = 0, 0
layer_list = self.forward_funcs[start:end]
```